6/Grains/src/HelloWorld.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System.Threading.Tasks;
+using Interfaces;
+using Orleans;
+
+namespace Grains
+{
+    public class HelloWorld : Orleans.Grain, IHelloWorld
+    {
+        public async Task<string> SayHello(string name, GrainCancellationToken grainCancellationToken)
+        {
+            string result = null;
+
+            if (!grainCancellationToken.CancellationToken.IsCancellationRequested)
+            {
+                result = $"Hello, {name} from {IdentityString}!";
+            }
+            else
+            {
+                result = $"Cancelled  from {IdentityString}.";
+            }
```

Comment:
Ah, if it's cancelled, the caller is most likely not interested in the response. We can safely return a null.